integration_tests/testsuite/test_logging.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python
+
+# Copyright 2016 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import unittest
+from retrying import retry
+
+import google.cloud.logging
+
+import test_util
+
+
+class TestLogging(unittest.TestCase):
+
+    def __init__(self, url, methodName='runTest'):
+        self._url = url + test_util.LOGGING_ENDPOINT
+        unittest.TestCase.__init__(self)
+
+    def runTest(self):
+        logging.debug('Posting to endpoint: {0}'.format(self._url))
+
+        payload = test_util._generate_logging_payload()
+        if test_util._post(self._url, payload) != 0:
+            return self.fail('Error encountered inside sample application!')
+
+        try:
+            client = google.cloud.logging.Client()
+            log_name = payload.get('log_name')
+            token = payload.get('token')
+
+            logging.info('log name is {0}, '
+                         'token is {1}'.format(log_name, token))
+
+            project_id = test_util._project_id()
+            FILTER = 'logName = projects/{0}/logs/' \
+                     'appengine.googleapis.com%2Fstdout'.format(project_id)
+
+            self._read_log(client, log_name, token, FILTER)
+            return 0
+        except Exception as e:
```

Comment:
You can get rid of try/except/fail in all of the runTest() methods, the test framework does it for you.  Unless you really want fine control over the message.